### PR TITLE
fixes #104

### LINKED ISSUE #104: SSL/TLS Not Supported
When attempting to submit to an https endpoint, the following error is generated:

```
Error: Protocol:https not supported.
    at Object.exports.request (http.js:1841:11)
    at FormData.submit (/some/random/path/node_modules/form-data/lib/form_data.js:312:20)
    at Request.<anonymous> (/some/random/path/lib/Engine.js:69:16)
    at Request.emit (events.js:95:17)
    at Request.onRequestResponse (/some/random/path/node_modules/request/request.js:1139:10)
    at ClientRequest.emit (events.js:95:17)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (http.js:1689:21)
    at HTTPParser.parserOnHeadersComplete [as onHeadersComplete] (http.js:122:23)
    at Socket.socketOnData [as ondata] (http.js:1584:20)
    at TCP.onread (net.js:528:27)
```

### PR DESCRIPTION
**Pull request by Zarlex**
fixes #104 and makes it possible to configure https by setting the port to the default ssl port 443

### CODE REVIEW

**Comment by alexindigo** [comment]
Thank you for the PR. And going through with the issue. 
Although it looks like changes to the test not reflecting changes to the code. Can you elaborate on the problem and add proper test for the changes (not modifying existing, but creating new one to assure backwards compatibility). Thank you.

File: test/integration/test-submit-http.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+// check params as object
+testRequest({protocol: 'http:', hostname: 'npmjs.org', pathname: '/'});
+
+testRequest({port: 80, hostname: 'npmjs.org', pathname: '/'});
+
+testRequest({port: 80, protocol: 'http:', hostname: 'npmjs.org', pathname: '/'});
+
+// --- Santa's little helpers
+
+function testRequest(params)
```
Comment (alexindigo) [inline test/integration/test-submit-http.js]: Codacy Issue found: ['testRequest' is defined but never used.](https://www.codacy.com/app/form-data/form-data/file/876417302/issues/source?bid=2805153&fileBranchId=2805189#l23)

Comment (alexindigo) [inline test/integration/test-submit-http.js]: It's Codacy bug, don't be alarmed. :)

**Comment by DylanPiercey** [comment]
@Zarlex this is an old PR and I haven't been following it. Do you mind letting me know if this is still an issue? https://github.com/form-data/form-data/issues/104 seems to suggest it has been resolved.

**Pull request closed without merging**